src/hooks/useUrlPagination.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useMemo } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+
+export type UseUrlPaginationInput = Pick<RouteComponentProps, 'location' | 'history'>;
+
+export type UseUrlPaginationOutput = {
+  currentPage: number;
+  setCurrentPage: (newPageIndex: number) => void;
+};
+
+export const PAGE_PARAM_NAME = 'page';
+
+const useUrlPagination = ({ location, history }: UseUrlPaginationInput): UseUrlPaginationOutput => {
+  const { search } = location;
+
+  useEffect(() => {
+    // Add page param to URl if none has been set
+    const searchParams = new URLSearchParams(search);
+
+    if (!searchParams.get(PAGE_PARAM_NAME)) {
+      // Note: although the pagination starts from 0, we make it start from 1 in the
```

Comment:
This shouldn't be necessary. We discussed handling this on the backend and versioned the endpoint to paginate with an initial page value of 1 on the transactions endpoint. 

We can do the same for proposals

Reply:
As of today, this isn't implemented so we need the front end to take care of it (like it currently does). We'll need to update this logic once the backend is updated.

As per the pagination, the implementation in this PR does work with a page number (you can spin up the app and check). That part I actually didn't change: that's how the current implementation is.